pkgpanda/__init__.py

```diff
@@ -9,11 +9,17 @@
 environment variables from the package.
 
 """
-import grp
+
+import platform
+is_windows = platform.system() is "Windows"
```

Comment:
This is what's failing the `teamcity/dcos/build/tox` check with `E402 module level import not at top of file`. This non-import line causes the linter to believe that the file is done declaring imports, so it throws an error when it sees the imports below. You can fix this by moving this line to follow all imports, and wrapping the `grp` and `pwd` imports in a `try` block:

```
try:
    import grp
except ImportError:
    pass
```

Then after the imports you can set `is_windows` and assert that we have the correct set of modules imported for the platform platform:

```
is_windows = platform.system() == 'Windows'


if not is_windows:
    expected_modules = ['grp', 'pwd']
    for module_name in expected_modules:
        assert module_name in sys.modules, 'Module {} was not imported'.format(module_name)
```

That will get you conditional imports while keeping all import declarations grouped together at the top of the file.

Reply:
OK. I'll try that.  That said, how would that handle the import of modules which are incompatible with each other but still present?  There are a few of those that I've seen, though I doubt that is an issue in this particular case.

Reply:
Can you clarify what you mean by "modules which are incompatible with each other but still present"? If you're trying to use one module instead of another when you're on Windows, you can do something like this:

```
import mod_nix
import mod_win

mod = mod_nix
if platform.system() == 'Windows':
    mod = mod_win

def main():
    mod.stuff()
```

In any case I'd bet that a refactor would help manage platform-specific imports once they get complicated. If we pushed platform-specific behavior down into `pkgpanda` submodules that encapsulated it well with a consistent API, we could do one conditional import in `pkgpanda/__init__.py` to pull in the right submodule for the platform. And then submodules can just assume they're on the platform they intend, with no need for elaborate import logic.